Provide 'booktabs' style for tables.

godbyk opened the pull request:
**This pull request is intended to elicit feedback.**

The [booktabs package](http://ctan.org/pkg/booktabs) provides nice rules and spacing for tables. Tables of this style are frequently used in academic journals.

A heavy rule is used at the top and bottom of the table. A medium-weight rule is separates the head row from the table contents.

One unresolved issue: centering the table by default. I almost always center my tables and I want them to only be as wide as necessary (not expand to consume 100% of the width available). This pull request centers the table, but it's offset slight—I think due to the gap between the main text block and the sidenotes area.

Future potential features:
- `\addlinespace` functionality to add additional spacing between a specific row. This is useful for lengthy tables. Providing periodic spacing improves readability without resorting to zebra backgrounds and the like.
- `\cmidrule`s where the width of the rule can be adjusted to provide gaps between adjacent rules.

daveliepmann commented:
This could be useful as an alternative table class—that is, leaving the default `table` styling untouched. Could you add an example in `index.html` that uses all your new classes?

godbyk committed:
Provides a `booktabs` class for styling tables.

The booktabs package <http://ctan.org/pkg/booktabs> provides nice rules
and spacing for tables. Tables of this style are frequently used in
academic journals.

A heavy rule is used at the top and bottom of the table.
A medium-weight rule is separates the head row from the table contents.

godbyk commented:
I've updated the pull request. The `booktabs` formatting is now applied only if you specify `<table class="booktabs">`. I also provided an example table and a brief description of the style.

daveliepmann commented:
Cool stuff. Thanks!

gsauthof commented:
In case anybody else is wondering why such table styling doesn't work anymore in the current CSS:

It was removed again in November, 2015:

commit cea451a1b6516901d500e33d564c7bff8cf16298
Author: Dave Liepmann <dave...
Date:   Tue Nov 17 18:29:12 2015 +0200

    Nix all table styles per ET
